android/app/src/main/java/org/uwaterloo/subletr/pages/createlisting/CreateListingPageView.kt

```diff
@@ -451,6 +536,104 @@ private val storeDateFormatISO: DateTimeFormatter = DateTimeFormatter.ISO_LOCAL_
 @OptIn(ExperimentalMaterial3Api::class)
 private val displayDateFormatter = DatePickerDefaults.dateFormatter(selectedDateSkeleton = "MM/dd/yyyy")
 
+@Composable
+fun CreateListingNumericalInputs(
+	modifier: Modifier = Modifier,
+	labelId: Int,
+	viewModelStream: BehaviorSubject<Int>,
+	uiStateValue: Int,
+) {
+	val numberPattern = remember { Regex("^\\d+\$") }
+
+	RoundedTextField(
+		modifier = modifier
+			.fillMaxWidth()
+			.border(
+				dimensionResource(id = R.dimen.xxxs),
+				textFieldBorderColor,
+				RoundedCornerShape(dimensionResource(id = R.dimen.xxxxl))
+			),
+		placeholder = {
+			Text(
+				text = stringResource(id = labelId),
+				color = secondaryTextColor,
+			)
+		},
+		label = {
+			Text(
+				text = stringResource(id = labelId),
+				color = secondaryTextColor,
+			)
+		},
+		keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number),
+		value = if (uiStateValue == 0) "" else uiStateValue.toString(),
+		onValueChange = {
+			if (it.isEmpty()) {
+				viewModelStream.onNext(0)
+			} else if (it.matches(numberPattern)) {
+				viewModelStream.onNext(it.toInt())
+			}
+		}
+	)
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun <T> CreateListingSelectionInput(
+	modifier: Modifier = Modifier,
+	dropdownItems: Array<T>,
+	labelId: Int,
+	selectedDropdownItem: T,
+	dropdownItemToString: @Composable (T) -> String,
+	setSelectedDropdownItem: (T) -> Unit,
+) {
```

Comment:
took inspiration from the wonderful `RoundedDropdown` and modified it a bit since there's some differences to the create listing dropdowns